ValueError in test/petab/test_amici_objective.py::test_error_leastsquares_with_ssigma

https://github.com/ICB-DCM/pyPESTO/actions/runs/6945543894/job/18895149022

```python
_____________________ test_error_leastsquares_with_ssigma ______________________

    def test_error_leastsquares_with_ssigma():
        model_name = "Zheng_PNAS2012"
        petab_problem = petab.Problem.from_yaml(
            os.path.join(models.MODELS_DIR, model_name, model_name + ".yaml")
        )
        petab_problem.model_name = model_name
        importer = pypesto.petab.PetabImporter(petab_problem)
        obj = importer.create_objective()
        problem = importer.create_problem(obj)
    
        optimizer = pypesto.optimize.ScipyOptimizer(
            'ls_trf', options={'max_nfev': 50}
        )
        with pytest.raises(RuntimeError):
>           optimize.minimize(
                problem=problem,
                optimizer=optimizer,
                n_starts=1,
                options=optimize.OptimizeOptions(allow_failed_starts=False),
                progress_bar=False,
            )

test/petab/test_amici_objective.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pypesto/optimize/optimize.py:156: in minimize
    ret = engine.execute(tasks, progress_bar=progress_bar)
pypesto/engine/single_core.py:34: in execute
    results.append(task.execute())
pypesto/optimize/task.py:59: in execute
    optimizer_result = self.optimizer.minimize(
pypesto/optimize/optimizer.py:176: in wrapped_minimize
    result = minimize(
pypesto/optimize/optimizer.py:144: in wrapped_minimize
    result = minimize(
pypesto/optimize/optimizer.py:85: in wrapped_minimize
    result = minimize(
pypesto/optimize/optimizer.py:333: in minimize
    res = scipy.optimize.least_squares(
.tox/petab/lib/python3.9/site-packages/scipy/optimize/_lsq/least_squares.py:940: in least_squares
    result = trf(fun_wrapped, jac_wrapped, x0, f0, J0, lb, ub, ftol, xtol,
.tox/petab/lib/python3.9/site-packages/scipy/optimize/_lsq/trf.py:123: in trf
    return trf_bounds(
.tox/petab/lib/python3.9/site-packages/scipy/optimize/_lsq/trf.py:316: in trf_bounds
    S, _ = qr(S, mode='economic')
.tox/petab/lib/python3.9/site-packages/scipy/linalg/_decomp_qr.py:128: in qr
    a1 = numpy.asarray_chkfinite(a)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = array([[nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
    ...an],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan]])
dtype = None, order = None

    @set_module('numpy')
    def asarray_chkfinite(a, dtype=None, order=None):
        """Convert the input to an array, checking for NaNs or Infs.
    
        Parameters
        ----------
        a : array_like
            Input data, in any form that can be converted to an array.  This
            includes lists, lists of tuples, tuples, tuples of tuples, tuples
            of lists and ndarrays.  Success requires no NaNs or Infs.
        dtype : data-type, optional
            By default, the data-type is inferred from the input data.
        order : {'C', 'F', 'A', 'K'}, optional
            Memory layout.  'A' and 'K' depend on the order of input array a.
            'C' row-major (C-style),
            'F' column-major (Fortran-style) memory representation.
            'A' (any) means 'F' if `a` is Fortran contiguous, 'C' otherwise
            'K' (keep) preserve input order
            Defaults to 'C'.
    
        Returns
        -------
        out : ndarray
            Array interpretation of `a`.  No copy is performed if the input
            is already an ndarray.  If `a` is a subclass of ndarray, a base
            class ndarray is returned.
    
        Raises
        ------
        ValueError
            Raises ValueError if `a` contains NaN (Not a Number) or Inf (Infinity).
    
        See Also
        --------
        asarray : Create and array.
        asanyarray : Similar function which passes through subclasses.
        ascontiguousarray : Convert input to a contiguous array.
        asfarray : Convert input to a floating point ndarray.
        asfortranarray : Convert input to an ndarray with column-major
                         memory order.
        fromiter : Create an array from an iterator.
        fromfunction : Construct an array by executing a function on grid
                       positions.
    
        Examples
        --------
        Convert a list into an array.  If all elements are finite
        ``asarray_chkfinite`` is identical to ``asarray``.
    
        >>> a = [1, 2]
        >>> np.asarray_chkfinite(a, dtype=float)
        array([1., 2.])
    
        Raises ValueError if array_like contains Nans or Infs.
    
        >>> a = [1, 2, np.inf]
        >>> try:
        ...     np.asarray_chkfinite(a)
        ... except ValueError:
        ...     print('ValueError')
        ...
        ValueError
    
        """
        a = asarray(a, dtype=dtype, order=order)
        if a.dtype.char in typecodes['AllFloat'] and not np.isfinite(a).all():
>           raise ValueError(
                "array must not contain infs or NaNs")
E           ValueError: array must not contain infs or NaNs

.tox/petab/lib/python3.9/site-packages/numpy/lib/function_base.py:630: ValueError
```

--- a/test/petab/test_amici_objective.py
+++ b/test/petab/test_amici_objective.py
@@ -49,6 +49,7 @@ def test_add_sim_grad_to_opt_grad():
     assert np.allclose(expected, opt_grad)
 
 
+@pytest.mark.flaky(reruns=3)
 def test_error_leastsquares_with_ssigma():
     model_name = "Zheng_PNAS2012"
     petab_problem = petab.Problem.from_yaml(

--- a/test/petab/test_amici_objective.py
+++ b/test/petab/test_amici_objective.py
@@ -49,7 +49,7 @@ def test_add_sim_grad_to_opt_grad():
     assert np.allclose(expected, opt_grad)
 
 
-@pytest.mark.flaky(reruns=3)
+@pytest.mark.flaky(reruns=2)
 def test_error_leastsquares_with_ssigma():
     model_name = "Zheng_PNAS2012"
     petab_problem = petab.Problem.from_yaml(
@@ -58,7 +58,9 @@ def test_error_leastsquares_with_ssigma():
     petab_problem.model_name = model_name
     importer = pypesto.petab.PetabImporter(petab_problem)
     obj = importer.create_objective()
-    problem = importer.create_problem(obj)
+    problem = importer.create_problem(
+        obj, startpoint_kwargs={'check_fval': True, 'check_grad': True}
+    )
 
     optimizer = pypesto.optimize.ScipyOptimizer(
         'ls_trf', options={'max_nfev': 50}

